pkg/plugin/v2/init.go

```diff
@@ -0,0 +1,185 @@
+`, ctx.CommandName)
+}
+
+func (p *initPlugin) BindFlags(fs *pflag.FlagSet) {
+	fs.BoolVar(&p.skipGoVersionCheck, "skip-go-version-check",
+		false, "if specified, skip checking the Go version")
+
+	// dependency args
+	fs.BoolVar(&p.fetchDeps, "fetch-deps", true, "ensure dependencies are downloaded")
+
+	// deprecated dependency args
+	fs.BoolVar(&p.dep, "dep", true, "if specified, determines whether dep will be used.")
+	p.depFlag = fs.Lookup("dep")
+	fs.StringArrayVar(&p.depArgs, "depArgs", nil, "additional arguments for dep")
+
+	if err := fs.MarkDeprecated("dep", "use the fetch-deps flag instead"); err != nil {
+		log.Printf("error to mark dep flag as deprecated: %v", err)
+	}
+	if err := fs.MarkDeprecated("depArgs", "will be removed with version 1 scaffolding"); err != nil {
+		log.Printf("error to mark dep flag as deprecated: %v", err)
+	}
+
+	// boilerplate args
+	fs.StringVar(&p.license, "license", "apache2",
+		"license to use to boilerplate, may be one of 'apache2', 'none'")
+	fs.StringVar(&p.owner, "owner", "", "owner to add to the copyright")
+
+	// project args
+	p.config = config.New(config.DefaultPath)
+	fs.StringVar(&p.config.Repo, "repo", "", "name to use for go module (e.g., github.com/user/repo), "+
+		"defaults to the go package of the current working directory.")
+	fs.StringVar(&p.config.Domain, "domain", "my.domain", "domain for groups")
+}
+
+func (p *initPlugin) Run() error {
+	return cmdutil.Run(p)
+}
+
+func (p *initPlugin) SetVersion(v string) {
+	if p.config == nil {
+		p.config = config.New(config.DefaultPath)
+	}
+	p.config.Version = v
+}
+
+func (p *initPlugin) LoadConfig() (*config.Config, error) {
+	_, err := config.Read()
+	if err == nil || os.IsExist(err) {
+		return nil, errors.New("already initialized")
+	}
+
+	return p.config, nil
+}
+
+func (p *initPlugin) Validate(c *config.Config) error {
+	// Requires go1.11+
+	if !p.skipGoVersionCheck {
+		if err := internal.ValidateGoVersion(); err != nil {
+			return err
+		}
+	}
+
+	// Check if the project name is a valid namespace according to k8s
+	dir, err := os.Getwd()
+	if err != nil {
+		return fmt.Errorf("error to get the current path: %v", err)
+	}
+	projectName := filepath.Base(dir)
+	if err := internal.IsDNS1123Label(strings.ToLower(projectName)); err != nil {
+		return fmt.Errorf("project name (%s) is invalid: %v", projectName, err)
+	}
+
+	// Try to guess repository if flag is not set
+	if c.Repo == "" {
+		repoPath, err := internal.FindCurrentRepo()
+		if err != nil {
+			return fmt.Errorf("error finding current repository: %v", err)
+		}
+		c.Repo = repoPath
+	}
+
+	return nil
+}
+
+func (p *initPlugin) GetScaffolder(c *config.Config) (scaffold.Scaffolder, error) { // nolint:unparam
+	return scaffold.NewInitScaffolder(c, p.license, p.owner), nil
+}
+
+func (p *initPlugin) PostScaffold(c *config.Config) error {
+	if (!p.depFlag.Changed && p.fetchDeps) || (p.depFlag.Changed && p.dep) {
```

Comment:
Nice catch, I missed that when refactoring, probably this should be fixed outside of the plugin implementation too.